src/IISIntegration/src/Microsoft.AspNetCore.Server.IIS/Core/IISServerSetupFilter.cs

```diff
@@ -28,9 +33,21 @@ public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
                     throw new InvalidOperationException("Application is running inside IIS process but is not configured to use IIS server.");
                 }
 
+                app.ServerFeatures.Set<IServerAddressesFeature>(new ServerAddressesFeature(_bindings));
```

Comment:
For consistency with the other servers the IServerAddressesFeature should created in the Server constructor and then populated in Server.StartAsync.